src/Aspire.Hosting.Azure.Provisioning/Provisioners/BicepProvisioner.cs

```diff
@@ -0,0 +1,148 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System.Diagnostics;
+using System.Text;
+using System.Text.Json.Nodes;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Dcp.Process;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Aspire.Hosting.Azure.Provisioning;
+
+internal sealed class BicepProvisioner(ILogger<BicepProvisioner> logger) : AzureResourceProvisioner<AzureBicepResource>
+{
+    public override bool ConfigureResource(IConfiguration configuration, AzureBicepResource resource)
+    {
+        var section = configuration.GetSection($"Azure:Deployments:{resource.Name}");
+
+        if (!section.Exists())
+        {
+            return false;
+        }
+
+        foreach (var item in section.GetSection("Outputs").GetChildren())
```

Comment:
What happens if you change the template? You might need to store a hash of the bicep file here somewhere so you can do change detection, and if that occurs execute the deployment again (you would need to nuke all the output values in that case).